### ISSUE TITLE
Question : Can't get table rows to produce content

### ISSUE THREAD

**Issue by chrismalek** [willoma/bulma-gomponents#1]
I am sure this is something simple I am failing to see.

I am creating a table result from a database query.  I am trying to create a bulma table but for some reason the body rows are NOT coming back.

```go
func (service *AdminUserService) uiAdminBulmaUserSearchResultTable(searchResults []database.User) *b.Element {

	if len(searchResults) == 0 {
		return el.Div("No Search Results Found")
	} else {
		rows := []any{}

		for _, user := range searchResults {

			rows = append(rows, b.Row(
				b.Cell(fmt.Sprintf("%d", user.ID)),
				b.Cell(user.Email),
				b.Cell(user.FirstName),
				b.Cell(user.LastName),
				b.Cell(user.AuthProvider),
				b.Cell(user.Created.Local().String()),
				b.Cell(user.UpdatedAt.Local().String()),
			),
			)

		}

		bTable := b.Table(
			b.Bordered,
			b.Striped,
			b.HeadRow(
				b.Cell("ID"),
				b.Cell("Email"),
				b.Cell("First Name"),
				b.Cell("Last Name"),
				b.Cell("Auth Provider"),
				b.Cell("Created"),
				b.Cell("Updated"),
			),
			rows,
		)
		return bTable

	}

}
```

The `rows` here is not actually doing anything. I added some debug statements and my `row` has two elements.

However the only output is.

```html

<table class="table is-bordered is-striped">
    <thead>
        <tr>
            <th>ID</th>
            <th>Email</th>
            <th>First Name</th>
            <th>Last Name</th>
            <th>Auth Provider</th>
            <th>Created</th>
            <th>Updated</th>
        </tr>
    </thead>
</table>

```

This has something to do with the type of the rows.

I see in the library that there is some type checking, and if the inbound arrays to some of the functions are not what is expected, it seems just to drop the data given. I am not sure why this is. 

I dug through the library looking for clues but I am failing to see what I am missing.

**Comment by willoma** [comment]
This is because you are providing the rows slice as a single argument. You may either create the `bTable` variable before the  `for` loop and then use `bTable.With(b.Row(XXX))` inside the loop, or use `b.Table(XXX).Withs(rows)`.

I will update the doc on this subject (see #2)

**Comment by chrismalek** [comment]
Yes, I did try what you suggested. At least, I think I did, and I may have a blatant bug in my code. 

This does NOT work.

```go
func (service *AdminUserService) uiAdminBulmaUserSearchResultTable(searchResults []database.User) *b.Element {

	if len(searchResults) == 0 {
		return el.Div("No Search Results Found")
	} else {

		bTable := b.Table(
			b.Bordered,
			b.Striped,
			b.HeadRow(
				b.Cell("ID"),
				b.Cell("Email"),
				b.Cell("First Name"),
				b.Cell("Last Name"),
				b.Cell("Auth Provider"),
				b.Cell("Roles"),
				b.Cell("Created"),
				b.Cell("Updated"),
			))

		for _, user := range searchResults {

			bRow := b.Row(
				b.Cell(fmt.Sprintf("%d", user.ID)),
				b.Cell(user.Email),
				b.Cell(user.FirstName),
				b.Cell(user.LastName),
				b.Cell(user.AuthProvider),
				b.Cell(user.Roles.Value()),
				b.Cell(user.Created.Local().String()),
				b.Cell(user.UpdatedAt.Local().String()),
			)

			bTable.With(bRow)

		}

		return bTable

	}

}

```

The row cells using the code above would never take.

I ended up just NOT using the `b.row` and going back to the `el.*` and this worked fine.



```go
func (service *AdminUserService) uiAdminBulmaUserSearchResultTable(searchResults []database.User) *b.Element {

	if len(searchResults) == 0 {
		return el.Div("No Search Results Found")
	} else {

		bTable := b.Table(
			b.Bordered,
			b.Striped,
			b.HeadRow(
				b.Cell("ID"),
				b.Cell("Email"),
				b.Cell("First Name"),
				b.Cell("Last Name"),
				b.Cell("Auth Provider"),
				b.Cell("Roles"),
				b.Cell("Created"),
				b.Cell("Updated"),
			))

		for _, user := range searchResults {

			bTable.With(el.Tr(
				el.Td(fmt.Sprintf("%d", user.ID)),
				el.Td(user.Email),
				el.Td(user.FirstName),
				el.Td(user.LastName),
				el.Td(user.AuthProvider),
				el.Td(user.Roles.Value()),
				el.Td(user.Created.Local().String()),
				el.Td(user.UpdatedAt.Local().String()),
			))

		}

		return bTable

	}

}
```


I am still trying to wrap my head around the library as the difference between the el and b 

```go
b "github.com/willoma/bulma-gomponents"
el "github.com/willoma/bulma-gomponents/el"

```

**Comment by willoma** [comment]
oops my bad,  it would indeed NOT work with `.With` nor with `.Withs`, the approach I took is not okay with this. I will improve how we can deal with it, which may end up in your first try being okay (ie. providing a slice of `any` as an argument).

I'm going to leave this issue open until it is improved.

The following approach would work for now: create a `bTableArgs := []any{ ........ }`, append the `b.Row` elements to that slice in the loop and return `b.Table(bTableArgs...)`.

Your approach that works fine has a drawback: the rows are not placed in a `tbody` element.

-----

`b` is the collection of functions and values that correspond to Bulma elements, components, etc.

`el` is a collection of helpers for standard HTML elements (ie. nothing specific to bulma) that wrap elements from "github.com/maragudk/gomponents/html" into `b.Element` instances, which would then accept attributes from bulma-gomponents. For instance, the two following lines return identical HTML:

```go
html.Div(html.Class("has-text-primary has-background-danger-light"), gomponents.Text("Hello"))
el.Div(b.Primary.Text(), b.Danger.BackgroundLight(),  "Hello")
```

**Comment by willoma** [comment]
FYI I have started working on a change, applied to most functions in `b`, that will allow providing `[]any` as arguments, the functions will then include the contained values (recursively).

**Comment by willoma** [comment]
FYI I have started working on a feature to allow achieving what you need, that will be  applied to most functions in `b`.

**Comment by chrismalek** [comment]
I slept on this. I sort of think that all (or most) the Bulma function returns should be a github.com/maragudk/gomponents Node or have a property or something to get at a github.com/maragudk/gomponents node.  I just started to use github.com/maragudk/gomponents and I am still wrapping my head around the structure of the library.

**Comment by willoma** [comment]
All functions (except classes, etc) return a `b.Element`, which satisfies the `gomponents.Node` interface. Moreover, all these functions accept `gomponent.Node` children :-)

I will improve the documentation on this aspect

**Comment by willoma** [comment]
I have already pushed modifications that allow providing `[]any` children like you first tried, and I am working on a way to provide those children with the `.With` function on all elements/components/etc too, as I wrongly suggested. I will tag a new release when it is done.

**Comment by chrismalek** [comment]
> All functions (except classes, etc) return a `b.Element`, which satisfies the `gomponents.Node` interface. 


Ok..let me look at that interface a little closer. Like I said I am still wrapping my head around that library. Thanks for the improvements. I added myself as a watcher on the project.

**Comment by willoma** [comment]
Ability to provide children in `[]any` slices, or in the `With` function, is implemented in [v0.3.0](https://github.com/willoma/bulma-gomponents/releases/tag/v0.3.0)! Documentation has also been improved, with details on how to use the lib and how to interact with gomponents when needed.
